### PR TITLE
APP-1531 Contracts display refactor

### PR DESCRIPTION
**Pull request by tarikstafford**
## [APP-1531]

- Removed `Contract Filter` instead we have two arrays of contracts in the `ContractStore` now.
- Created a new `TerminatedContracts` view to reduce noise.

## Checklist

- [ ] Dark/light mode verification
- [ ] Landscape verification
- [ ] iPad verification
- [ ] iPod/iPhone 12 mini/iPhone SE verification

[APP-1531]: https://hedvig.atlassian.net/browse/APP-1531?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Projects/Contracts/Sources/Contracts.swift [hunk 1]
```diff
@@ -85,6 +35,7 @@ extension Contracts: View {
             fetch()
         }
         .trackOnAppear(hAnalyticsEvent.screenView(screen: .insurances))
+        .navigationBarTitle(L10n.InsurancesTab.title)
```
Comment (sampettersson) [inline Projects/Contracts/Sources/Contracts.swift:38]: This one usually leads to the title being set too late and it looks off, why did you move this away from the Journey?

File: Projects/Contracts/Sources/Contracts.swift [hunk 2]
```diff
@@ -8,64 +8,14 @@ import hCore
...
-    let filter: ContractFilter
-
-    public init(
-        filter: ContractFilter
-    ) {
-        self.filter = filter
-    }
+    
+    @State
+    var navigationController: UINavigationController?
```
Comment (sampettersson) [inline Projects/Contracts/Sources/Contracts.swift:16]: This is not used right?

**Review by sampettersson — CHANGES_REQUESTED** [review]

File: Projects/Contracts/Sources/View/ContractTable.swift [hunk 3]
```diff
@@ -9,82 +9,72 @@ import hCoreUI
...
-        case .active:
-            return state
-                .contractBundles
-                .flatMap { $0.contracts }
-        case .terminated:
-            return state.contracts.filter { contract in
-                contract.currentAgreement?.status == .terminated
-            }
-        case .none: return []
+        if contracts.isEmpty {
```
Comment (SatishBV) [inline Projects/Contracts/Sources/View/ContractTable.swift:20]: Maybe this? 
```suggestion
        self.contracts = contracts.isEmpty ? store.state.terminatedContracts : contracts
```

File: Projects/Contracts/Sources/View/ContractTable.swift [hunk 4]
```diff
@@ -9,82 +9,72 @@ import hCoreUI
...
-                    getContractsToShow(for: state, filter: filter.nonemptyFilter(state: state))
-                }
-            ) { contracts in
-                ForEach(contracts, id: \.id) { contract in
-                    ContractRow(id: contract.id)
-                        .fixedSize(horizontal: false, vertical: true)
-                        .padding(.top, 15)
-                        .transition(.slide)
-                }
+            ForEach(contracts, id: \.id) { contract in
```
Comment (SatishBV) [inline Projects/Contracts/Sources/View/ContractTable.swift:33]: Does removing the `PresentableStoreLens` not have any affect when polling for contracts?

File: Projects/Contracts/Sources/View/ContractTable.swift [hunk 5]
```diff
@@ -9,82 +9,72 @@ import hCoreUI
...
+        
+        hSection {
+            PresentableStoreLens(
+                ContractStore.self,
+                getter: { state in
+                    return state.terminatedContracs
+                }
+            ) { terminatedContracts in
+                if !terminatedContracts.isEmpty {
+                    hSection(header: hText(L10n.InsurancesTab.moreTitle)) {
```
Comment (SatishBV) [inline Projects/Contracts/Sources/View/ContractTable.swift:55]: Maybe move this view to a new property and avoid this indent gore? Not super important but felt it would be much more clearer to make sense of what is going on here

**Review by SatishBV — COMMENTED** [review]
Super nice this is 🚀

File: Projects/Contracts/Sources/View/ContractTable.swift (hunk 4 above)
Comment (tarikstafford) [reply Projects/Contracts/Sources/View/ContractTable.swift:33]: Not that I am aware of. The polling should update the `state` which is received in the `onReceive` block, which then updates the local `state` value.

File: Projects/Contracts/Sources/View/ContractTable.swift (hunk 5 above)
Comment (tarikstafford) [reply Projects/Contracts/Sources/View/ContractTable.swift:55]: Yeah good call!

File: Projects/Contracts/Sources/Contracts.swift (hunk 1 above)
Comment (tarikstafford) [reply Projects/Contracts/Sources/Contracts.swift:38]: Can move it back, originally I was going to reuse the `ContractsTable` for `Terminated` as well but now I created a new view. So I can just move it back 👯

**Review by SatishBV — APPROVED** [review]

File: Projects/hCore/Sources/ScrollView+Inspect.swift [hunk 6]
```diff
@@ -65,6 +65,83 @@ struct ViewIntrospector<ViewType: UIView>: UIViewRepresentable {
     }
 }
 
+class IntrospectionViewController: UIViewController {
```
Comment (sampettersson) [inline Projects/hCore/Sources/ScrollView+Inspect.swift:68]: Remove if you didn't end up using this maybe?

**Review by sampettersson — APPROVED** [review]

File: Projects/hCore/Sources/ScrollView+Inspect.swift (hunk 6 above)
Comment (tarikstafford) [reply Projects/hCore/Sources/ScrollView+Inspect.swift]: 🆗

**Review by sampettersson — APPROVED** [review]
